packages/framework/src/Model/Order/OrderFacade.php

```diff
@@ -491,9 +500,192 @@ protected function fillOrderItems(Order $order, OrderPreview $orderPreview)
     {
         $locale = $this->domain->getDomainConfigById($order->getDomainId())->getLocale();
 
-        $order->fillOrderProducts($orderPreview, $this->orderItemFactory, $this->numberFormatterExtension, $locale);
-        $order->fillOrderPayment($this->paymentPriceCalculation, $this->orderItemFactory, $orderPreview->getProductsPrice(), $locale);
-        $order->fillOrderTransport($this->transportPriceCalculation, $this->orderItemFactory, $orderPreview->getProductsPrice(), $locale);
-        $order->fillOrderRounding($this->orderItemFactory, $orderPreview->getRoundingPrice(), $locale);
+        $this->fillOrderProducts($order, $orderPreview, $locale);
+        $this->fillOrderPayment($order, $orderPreview, $locale);
+        $this->fillOrderTransport($order, $orderPreview, $locale);
+        $this->fillOrderRounding($order, $orderPreview, $locale);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Order $order
+     * @param \Shopsys\FrameworkBundle\Model\Order\Preview\OrderPreview $orderPreview
+     * @param string $locale
+     */
+    protected function fillOrderProducts(Order $order, OrderPreview $orderPreview, $locale)
+    {
+        $quantifiedItemPrices = $orderPreview->getQuantifiedItemsPrices();
+        $quantifiedItemDiscounts = $orderPreview->getQuantifiedItemsDiscounts();
+
+        foreach ($orderPreview->getQuantifiedProducts() as $index => $quantifiedProduct) {
+            $product = $quantifiedProduct->getProduct();
+            if (!$product instanceof Product) {
```

Comment:
Should this always be an instance of product - or not? 
If it can be null for example, then function `get_class()` returns unexpected result...

Reply:
you are right, it will always be an instance of Product entity, I will remove the if statement :+1: